src/components/ScholarshipAmountField.jsx

```diff
@@ -27,58 +27,50 @@ function ScholarshipAmountField(props) {
   const classes = useStyles();
   const { labelStyle, formik } = props;
 
-  function updateAmount(min, max) {
-    formik.setFieldValue('amount.min', min, true);
-    formik.setFieldValue('amount.max', max, true);
-  }
-
-  function onTypeChange(e) {
-    formik.setFieldValue('amount.type', e.target.value, true);
-  }
-
   const amountType = formik.values.amount.type;
-  const minAmount = formik.values.amount.min;
-  const maxAmount = formik.values.amount.max;
   const minError = getIn(formik.errors, 'amount.min');
   const maxError = getIn(formik.errors, 'amount.max');
 
-  function formatHelperText() {
-    if (!minError && !maxError) return;
-    if (getIn(formik.touched, 'amount.min') && minError) return minError;
-    if (getIn(formik.touche, 'amount.max') && maxError) return maxError;
+  let helperText = '';
+  if (
+    getIn(formik.touched, 'amount.min') ||
+    getIn(formik.touched, 'amount.max')
+  ) {
+    helperText = minError || maxError;
   }
 
   const inputFields = {};
   inputFields[AmountType.Varies] = (
     <>
       <AmountTextField
         error={Boolean(minError)}
-        value={minAmount || ''}
+        value={formik.values.amount.min || ''}
         onChange={(e) => {
           const val = parseInt(e.target.value, 10);
-          updateAmount(val || 0, maxAmount);
+          formik.setFieldValue('amount.min', val || 0, true);
         }}
       />
 
       <RemoveIcon className={classes.dash} />
 
       <AmountTextField
         error={Boolean(maxError)}
-        value={maxAmount || ''}
+        value={formik.values.amount.max || ''}
         onChange={(e) => {
           const val = parseInt(e.target.value, 10);
-          updateAmount(minAmount, val || 0);
+          formik.setFieldValue('amount.max', val || 0, true);
```

Comment:
2 things here:
1. It looks like `true` is the default value for `shouldValidate` according to [docs](https://formik.org/docs/api/formik#setfieldvalue-field-string-value-any-shouldvalidate-boolean--void) so it can be left out. If you don't want to validate on any text field change you can explicitly pass `false`. 
2. I just noticed the boolean is a bit confusing so maybe add a comment stating what it is. E.g.
```suggestion
          formik.setFieldValue('amount.max', val || 0, /* shouldValidate = */ true);
```

Same comment above for min and below for type.

Reply:
That is an interesting way of writing comments 🤔. I will make these changes in another PR.